### PR TITLE
vendor awsesproxy image

### PR DESCRIPTION
**Pull request by pgvishnuram**
## Summary of changes

We are vendoring aws es proxy to astronomer ap-vendor repo to manage and maintain external services.

## Images are updated by this PR

- [ ] alertmanager
- [ ] blackbox-exporter
- [ ] configmap-reloader
- [ ] curator
- [ ] elasticsearch
- [ ] elasticsearch-exporter
- [ ] fluentd
- [ ] git-sync
- [ ] grafana
- [ ] keda
- [ ] keda-metrics-apiserver
- [ ] kibana
- [ ] kube-state
- [ ] kubed
- [ ] nats-server
- [ ] nats-streaming
- [ ] nats-exporter
- [ ] nginx
- [ ] nginx-es
- [ ] node-exporter
- [ ] openresty
- [ ] pgbouncer
- [ ] pgbouncer-exporter
- [ ] postgresql
- [ ] postgres-exporter
- [ ] prometheus
- [ ] redis
- [ ] registry
- [ ] statsd-exporter

## Checklist

Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.

- [X] version.txt is updated in the changed image(s)

If a security scan fails for an unchanged image...

- [ ] a fix has been applied OR...
- [X] an [issue]() has been created
https://github.com/astronomer/issues/issues/3820

If adding a new image:

- [X] the directory has the same name you intend it to be published as, less a preceding "ap-"
- [X] the directory includes a file "version.txt", with version matching the underlying software version
- [X] the file .github/PULL_REQUEST_TEMPLATE.md is updated to include the image in the checklist
- [X] execute the script .circleci/generate_circleci_config.py, commit changes to .circleci/config.yml

### CODE REVIEW

File: awsesproxy/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+FROM golang:${GO_VERSION}-alpine AS builder
+
+ENV AWSESPROXY_DIR /go/src/github.com/abutaha/aws-es-proxy
+
+RUN set -ex \
+    && apk add --no-cache make git file
+
+WORKDIR $AWSESPROXY_DIR
+
+RUN git clone https://github.com/astronomer/aws-es-proxy.git $AWSESPROXY_DIR && git checkout 4b359c78b9052c4cb8145cb92c7d560b99512043
```
Comment (danielhoherd) [inline awsesproxy/Dockerfile:25]: - This sha does not match the version tag 1.3 <https://github.com/astronomer/aws-es-proxy/releases/tag/v1.3>
- There is no `cd` command after `clone`
```suggestion
RUN git clone https://github.com/astronomer/aws-es-proxy.git $AWSESPROXY_DIR && \
    cd $AWSESPROXY_DIR && \
    git checkout v1.3
```

**Review by danielhoherd — CHANGES_REQUESTED** [review]
Need to checkout the version tag, and need to cd. I added a suggestion with these changes.

File: awsesproxy/Dockerfile (hunk 1 above)
Comment (pgvishnuram) [reply awsesproxy/Dockerfile:25]: we not using the exact v1.3 tag that does not have go1.17 support - i have raised a PR to upstream , still not yet merged

Comment (pgvishnuram) [reply awsesproxy/Dockerfile:25]: it clone inside that directory - so cd int to that dir  is not needed

**Review by danielhoherd — APPROVED** [review]
It bugs me that the version isn't semver, but that's what the upstream uses, so 🤷‍♂️

File: awsesproxy/Dockerfile (hunk 1 above)
Comment (andriisoldatenko) [inline awsesproxy/Dockerfile:25]: @pgvishnuram i know you already merged it, but it make sense to explain why do we want checkout this specific commit. Example you can find https://github.com/astronomer/ap-vendor/blob/main/registry/Dockerfile#L32-L35

**Pull request merged**